### ISSUE TITLE
Expand / Add definition around route_type 

### ISSUE THREAD

**Issue by e-lo** [google/transit#310]
## Customer Need

*On behalf of Amtrak...*

*As a customer, I'd like to know (and potentially select a route by) if I can expect the bus route I've chosen to provide a comfortable amount of space to sit and some space for my personal items - so that I can evaluate if it is an appropriate route for me or if is worth any difference in costs.*

*As a customer, I'd like to know (and potentially select a route by) if I can expect the bus route I've chosen to requires me to navigate across vehicle rows where I cannot stand and sit, next to individuals that I may not know without the option to move during the trip - so that I can evaluate if it is an appropriate route for me (my ability and comfort level in being in close proximity to others) or if is worth any difference in costs.*

## Current Spec

[`routes.route_type`](https://gtfs.org/reference/static#routestxt) :
> Indicates the type of transportation used on a route. Valid options are:
> 
> 0 - Tram, Streetcar, Light rail. Any light rail or street level system within a metropolitan area.
> 1 - Subway, Metro. Any underground rail system within a metropolitan area.
> 2 - Rail. Used for intercity or long-distance travel.
> 3 - Bus. Used for short- and long-distance bus routes.
> 4 - Ferry. Used for short- and long-distance boat service.
> 5 - Cable tram. Used for street-level rail cars where the cable runs beneath the vehicle, e.g., cable car in San Francisco.
> 6 - Aerial lift, suspended cable car (e.g., gondola lift, aerial tramway). Cable transport where cabins, cars, gondolas or open chairs are suspended by means of one or more cables.
> 7 - Funicular. Any rail system designed for steep inclines.
> 11 - Trolleybus. Electric buses that draw power from overhead wires using poles.
> 12 - Monorail. Railway in which the track consists of a single rail or a beam

#### 1. Add `coach` as a `route_type` | Update `bus`

3:  Bus, 
- customers may be seated or standing
- differentiated from a van by customers being able to navigate across rows via an aisle that is standing-height

9: Coach, 
- a bus with premium features such as:  
  - more comfortable, dedicated seating. 
 -  May or may not include: space underneath and above the rider for luggage, individual climate control, on-board restrooms, entertainment, and connectivity.

#### 2. Add `van` as a `route_type`

???: van, a slightly larger auto where 
- a single or pair of doors provides access to multiple rows of seats, access to which is typically provided via a small side aisle or folding seats. 
- Interior ceiling is below typical standing height. 

## Previous discussion

#174 considered several additional `route_type` values and settled on adding one for monorail and one for trolleybus.  There is a lot of great discussion in the issue thread that is worth reviewing.
 - Intercity bus/coach was not added at the time due to concerns that it was:
    -  Potentially not backwards compatible because many feeds currently code intercity bus as bus
        - *I see this is non-breaking because old versions of the feed would still "make sense" to users and would not fail validation*
        - *Arguably trolleybus would have had the same issue?*
    -  Lack of specificity in the definition of intercity bus
        - *I think we can get to a good definition.*

I am unaware of any previous discussions about "vans" (vs autos) but would welcome them.  

## Relevant open proposals

The following proposals are also relevant:

- [GTFS-Vehicles]([bit.ly/gtfs-vehicles](http://bit.ly/gtfs-vehicles) which proposes very granular features available on various vehicle types.
- [GTFS-ModesAndNetworks](https://docs.google.com/document/d/1qWFZ8bYD0d4CVtNfXjzKyq3nBYEGIoDZREayseDnrok/edit#) 
  - Considers additional expansion of `route_type` and grouping of routes and route types into networks and modes.  
- [GTFS-OnDemand](https://bit.ly/gtfs-ondemand) 
  - Proposes the addition of an enum value for car/auto/van.

**Comment by skinkie** [comment]
I would suggest you to revive the extended route types #125 pull request. Because one-of adding does not make sense either, since everything has been specified before.

**Comment by skinkie** [comment]
My bad #279

**Comment by carlfredl** [comment]
Thanks for sharing this proposal, these are very good needs. I'm worried this proposal wouldn't actually would meet the user needs without better definitions.

By the definitions above, SDTJ Passport's [Mercedes Sprinters (image)](https://www.google.com/maps/uv?pb=!1s0x80d9499d84de8849%3A0xf7bbc7857819c58!3m1!7e115!4shttps%3A%2F%2Flh5.googleusercontent.com%2Fp%2FAF1QipPZYIEpHosiHKjWgffyOJuS2x00Og6spZgfeC0K%3Dw389-h160-k-no!5ssdtj%20-%20Google%20Search!15sCgIgAQ&imagekey=!1e10!2sAF1QipPZYIEpHosiHKjWgffyOJuS2x00Og6spZgfeC0K&hl=en&sa=X&ved=2ahUKEwj_ps2Wl9r2AhWFk4kEHRcaBV0Qoip6BAg7EAM) should be defined as a "Coach" since the people appear to be able to stand between the seats and they have "premium features," yet that might not solve the user needs for information about comfort (that is subjective and the user may have had something different in mind as a "Coach") and ability to change seats (that is not yet part of the definition).

We've been discussing additional bus granularity on and off for the last [decade](https://groups.google.com/g/gtfs-changes/c/6G7M1RJkXd8/m/luv2-DNGyDMJ). The challenge seems to be a lack of shared understanding of what we are standardizing (similar to Occupancies). I'd say we'll need more clear, widely accepted definitions of coach, bus, and van.

**Comment by rafapereirabr** [comment]
A quick two cents here, that perhaps it would also be a good idea to create a `BRT` code for `route_type`.

**Comment by e-lo** [comment]
> I'd say we'll need more clear, widely accepted definitions of coach, bus, and van.

I'd encourage us not to let the perfect in the way of "pretty good" here and solving a bunch of potential customer needs. The consequences of a customer being disappointed with SDTJ's "coach" definition seems pretty low (and it likely meets the category of a "mini coach"[ per wikipedia,](https://simple.wikipedia.org/wiki/Types_of_buses#Minicoach) at least)

**Comment by gcamp** [comment]
> I'd encourage us not to let the perfect in the way of "pretty good" here and solving a bunch of potential customer needs.

+1 

> A quick two cents here, that perhaps it would also be a good idea to create a BRT code for route_type.

Even if defining BRT is it's own problematic, this would be the route type addition that's the most useful to us

**Comment by e-lo** [comment]
Updates:
- Referenced #279

**Comment by carlfredl** [comment]
> Even if defining BRT is its own problematic, this would be the route type addition that's the most useful to us

Indeed, treating all buses equal until we can agree on how to classify BRT seems like a high cost, vs moving forward with some room for discretion that probably already exists between types like rail and light rail.

> I'd encourage us not to let the perfect in the way of "pretty good" here and solving a bunch of potential customer needs.

Agreed. I don't think we can fully solve these entire user stories, but we could probably improve their experience by improving expectations.

@gcamp and consumers, how is `route_type` actually used? What would it allow you to improve if we added `Van` or `BRT`?

**Comment by gcamp** [comment]
Specifically on BRT, we add that definition internally on some buses when we want to 
1) show that bus line on a global transit maps. BRT are often structural elements in a city and are worth showing next to rail lines
2) Nudge the trip planner to take the BRT vs a regular bus when possible.

**Comment by drewda** [comment]
One point of reference: for @transitland as a consumer of thousands of feeds, we handle the full list of extended route type. There are some producers who use them, especially outside of the US. However, we also map the extended route types back on to the spec's short list: https://github.com/interline-io/transitland-lib/blob/master/tl/enum/routetypes.go

I assume some other large consumers do this as well, just with a different style for which extended route type they map onto a smaller subset of their choosing. Based on that assumption, I think many consumers may be fine voting to extend the number of route_types. A longer list may be somewhat arbitrarily defined and used inconsistently by producers, but each consumer can decide which route_types warrant their own styling (or other treatment like @gcamp mentions) vs. consolidated into a smaller set of route_types.

**Comment by jgysland** [comment]
> Indeed, treating all buses as equal until we can agree on how to classify BRT seems like a high cost, vs moving forward with some room for discretion (this likely already exists between types like rail and light rail).

Also monorail vs rail. And "Cable Tram" versus "Tram, Streetcar, Light rail"? If these are distinctions worth making, then then adding a code for BRT seems like it should be uncontroversial.

>  BRT are often structural elements in a city and are worth showing next to rail lines

I only learned GTFS existed about an hour ago, after asking some friends who work for my local transit agency why BRT routes which have been put into service over the past couple years aren't drawn on Google Maps' transit view like LRT routes (but are drawn in other apps, e.g. Apple Maps and TransitApp).

Looking into it, I was initially disappointed when I saw the specification for `routes.txt` included no way to distinguish between BRT and non-BRT buses, but then delighted to see its an open-source project and there's an open issue suggesting that this be amended.

Of course for a change like this to be useful to me as a user, my local transit agency would have to update their feed to mark BRT routes with whatever the new `route_type` enum value, and Google Maps maintainers would have to implement changes to render it, but putting it in the spec seems like it would be a good start.

**Comment by miklcct** [comment]
I would like to see a distinction between a bus and a coach in the UK because they are regulated differently. 

Currently, the ITO World producer uses 200 for the coach routes. 

Also, in Hong Kong, there are minibus routes which use 16 to 19 seaters and are regulated differently to normal bus routes, and unregulated share taxi routes as well.

I suggest that we formalise a hierarchical structure, which is compatible with Google but without actually including Google's definition for the specific modes, such that distinctions can be made for local consumers while consumers not locally supporting the extended numbers can map it into base types. In other words, we should get the major route types (ending in the hundreds) into the spec.

**Comment by felixguendling** [comment]
As discussed in https://github.com/google/transit/pull/621 I would like to propose to standardize extended route types as they

- are in use all over Europe already anyway (because without them it doesn't really make sense)
- are supported by a lot of software
- map closely to NeTEx modes which is helpful in direct conversions

https://developers.google.com/transit/gtfs/reference/extended-route-types

**Comment by felixguendling** [comment]
cc @gcamp @leonardehrenfried @hbruch

**Comment by leonardehrenfried** [comment]
The pragmatic thing to do would be indeed to copy the extended route types into the specification and add new ones as the need arises.

If you wanted to be fancy you could frame the problem in terms of branding or vehicles.

GTFS has always been a highly pragmatic spec, maybe at the cost of some precision, so I would probably go with the first option, too.

**Comment by felixguendling** [comment]
> If you wanted to be fancy you could frame the problem in terms of branding or vehicles.

I would keep this separate for now.

I would prefer a "quick win" just copy & pasting in the spec what a lot of software supports already anyway.

**Comment by miklcct** [comment]
Unfortunately, although the idea is good, the categorisation and examples are not good to me.

For example, the 100-199 range mostly contains mainline railways, but there are examples where non-mainline railways are shown, such as the "Gatwick Shuttle" listed as an example is 108. It is clearly not a mainline railway. A hypothetical service which shuttle between Heathrow T2&3 and T&4 can be an example of it, while in my opinion, a new 4xx code should be used for these shuttles which are not part of the mainline railway network (and hence regulated differently and perceived differently by users).

Entur is currently using 101 for their "airport express" trains, which differ from true "high speed rail" like TGV or ICE.

I think we should define the hundred digit as the main mode in one pull request, and only copy the last two digits for services which are more clearly defined and less controversal (such as 714) in a separate pull request.

**Comment by felixguendling** [comment]
> Entur is currently using 101 for their "airport express" trains, which differ from true "high speed rail" like TGV or ICE.

I agree with you that this is a wrong use of the 101 route_type but just because those route_types have been used in the wrong way in the past somewhere, this doesn't mean they can be standardized. NeTEx has them already and being compatible to NeTEx as well as to the existing widespread use of extended route types is more important than the fact that their current usage is not 100% consistent. I also see wrong usage of the current GTFS non-extended route_types (1, 5, 6, 7 are good candidates). Basically all standards are sometimes used inconsistent in practice.

**Comment by tzujenchanmbd** [comment]
For reference - `route_type` count based on [Mobility Database](https://mobilitydatabase.org/)
https://docs.google.com/spreadsheets/d/1QLogCMNTQZMLo9kINjpRidFGGgRiJCnql4nQ7HjKsHk/edit?gid=0#gid=0
